pkg/scalers/nats_jetstream_scaler.go

```diff
@@ -241,6 +302,45 @@ func (s *natsJetStreamScaler) GetMetrics(ctx context.Context, metricName string,
 				if stream.Name == s.metadata.stream {
 					s.stream = stream
 				}
+
+				for _, consumer := range stream.Consumers {
+					if consumer.Name == s.metadata.consumer {
+						s.leaderName = consumer.Cluster.Leader
+					}
+				}
+			}
+		}
+	}
+
+	// Query the consumer leader pod, it has the accurate count.
+	if jsAccountResp.MetaCluster.ClusterSize > 1 {
+		NATSJetStreamLeaderEndpoint := s.getNATSJetStreamLeaderEndpoint()
+		leaderReq, err := http.NewRequestWithContext(ctx, http.MethodGet, NATSJetStreamLeaderEndpoint, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		leaderResp, err := s.httpClient.Do(leaderReq)
+		if err != nil {
+			s.logger.Error(err, "unable to access NATS JetStream consumer leader monitoring endpoint", "natsServerMonitoringLeaderEndpoint", NATSJetStreamLeaderEndpoint)
+			return []external_metrics.ExternalMetricValue{}, err
+		}
+
+		defer leaderResp.Body.Close()
+		var jsAccountLeaderResp jetStreamEndpointResponse
+		if err = json.NewDecoder(leaderResp.Body).Decode(&jsAccountLeaderResp); err != nil {
+			s.logger.Error(err, "unable to decode NATS JetStream consumer leader account details")
+			return []external_metrics.ExternalMetricValue{}, err
+		}
+
+		// Find and assign the leader stream that we are looking for.
+		for _, account := range jsAccountLeaderResp.Accounts {
+			if account.Name == s.metadata.account {
+				for _, stream := range account.Streams {
+					if stream.Name == s.metadata.stream {
+						s.leaderStream = stream
+					}
+				}
```

Comment:
maybe we should extract this code into a function because it's duplicated

Reply:
ya...I didn't want to upend the structure, but the only difference between the duplicate sections are the return values.

Reply:
yeah, probably a good idea, the only difference between the duplicate sections are the return values. maybe @goku321 will have some input.

Reply:
Hi @rayjanoka I know I've already delayed this so much but I've been dealing with some health issues recently and couldn't spend any time on the tests. Allow me some time and I'll try to speed things up. Really appreciate your patience.

Reply:
take care of yourself @goku321, no rush here.

Reply:
@goku321 absolutely, take care of yourself first

Reply:
this is resolved @JorTurFer